File: CLI/CKA/scripts/set_multi_node.py
import subprocess
import sys

def set_multi_node_cluster(control_plane_ip, control_plane_port, token, discovery_token_ca_cert_hash):
    # Join worker nodes to the control-plane node
    worker_nodes = ['worker1', 'worker2', 'worker3'] 

    for node in worker_nodes:
        # SSH into the worker node and execute the join command
        join_command = f'kubeadm join {control_plane_ip}:{control_plane_port} --token {token} --discovery-token-ca-cert-hash {discovery_token_ca_cert_hash}'
        ssh_command = f'ssh {node} "{join_command}"'
        subprocess.run(ssh_command, shell=True)

    # Add any other necessary steps to configure the cluster for multiple nodes

if __name__ == '__main__':
    if len(sys.argv) > 1 and sys.argv[1] == 'set_multi_node_cluster':
        set_multi_node_cluster('<control_plane_ip>', '<control_plane_port>', '<token>', '<discovery_token_ca_cert_hash>')
    else:
        # Default behavior
        install_dependencies()
        setup_cluster()
        run_application()
        perform_post_installation_tasks()

File: CLI/CKA/scripts/set_multi_node.go
package main

import (
	"fmt"
	"os"
	"os/exec"
)

func setMultiNodeCluster() {
	controlPlaneIP := "<control_plane_ip>"
	controlPlanePort := "<control_plane_port>"
	token := "<token>"
	discoveryTokenCACertHash := "<discovery_token_ca_cert_hash>"

	// Join worker nodes to the control-plane node
	workerNodes := []string{"worker1", "worker2", "worker3"}

	for _, node := range workerNodes {
		cmd := exec.Command("ssh", node, fmt.Sprintf("kubeadm join %s:%s --token %s --discovery-token-ca-cert-hash %s", controlPlaneIP, controlPlanePort, token, discoveryTokenCACertHash))
		cmd.Stdout = os.Stdout
		cmd.Stderr = os.Stderr
		err := cmd.Run()
		if err != nil {
			fmt.Printf("Error executing command on %s: %v\n", node, err)
		}
	}
}

func main() {
	if len(os.Args) > 1 && os.Args[1] == "set_multi_node_cluster" {
		setMultiNodeCluster()
	}
}
